include/max_element.hpp

```diff
@@ -21,11 +21,54 @@ namespace bit {
 
 
 
-// Status: to do
+// Status: complete
 template <class ForwardIt>
 constexpr bit_iterator<ForwardIt> max_element(bit_iterator<ForwardIt> first,
     bit_iterator<ForwardIt> last) {
-    (last);
+
+    using word_type = typename bit_iterator<ForwardIt>::word_type;
+    constexpr std::size_t num_digits = bit::binary_digits<word_type>::value; 
+
+    bit_iterator<ForwardIt> cursor = first;
+
+    // handle misaligned first word. for example ^00100^101
+    if (!_is_aligned_lsb(cursor)) {
+        word_type first_word = _padded_read(cursor, last, bit::bit0);
+        if (first_word > 0) {
+            std::size_t first_set_bit_position = _tzcnt(first_word);
+            return cursor + (first_set_bit_position - cursor.position());
+        }
+
+        if (_in_same_word(cursor, last) ||
+            _is_one_past_last_bit(last, cursor.base())) {
+            return first;
+        }
+
+        std::advance(cursor, num_digits - cursor.position());
+    }
+
+    // full words
+    while (!_in_same_word(cursor, last) ||
+        _is_one_past_last_bit(last, cursor.base())) {
+        word_type cur_word = *(cursor.base());
+        if (cur_word > 0) {
+            std::size_t cur_set_bit_position = _tzcnt(cur_word);
+            return cursor + (cur_set_bit_position - cursor.position());
+        }
+        std::advance(cursor, num_digits);
+    }
+
+    if (cursor == last) {
+        return first;
+    }
+
+    // final, incomplete word
+    word_type last_word = _padded_read(cursor, last, bit::bit0);
+    if (last_word > 0) {
+        word_type last_set_bit_position = _tzcnt(last_word);
+        return cursor + (last_set_bit_position - cursor.position());
```

Comment:
Won't `cursor.position()` always be `0` here? We align it on line 47, and then only add whole words to it.